### ISSUE TITLE
App Shows blank Screen when in Production Mode

### ISSUE THREAD

**Issue by ayman-rahmon** [expo/expo#15238]
### Summary

The app works perfectly when i run it with expo start but as soon as i run it with expo start --no-dev --minify  i get a white screen after the splash screen no matter what i do (same happen in apk stand alone builds as well).     when i get this white screen i also get no errors or output in the terminal whatsoever just a blank white screen with nothing to go on to fix the problem .

i was running the app on expo SDK 41 and i upgraded to SDK 43 thinking it would solve the problem but it did nothing of the sort (after fixing all of the dependencies and upgrading i came back to the same problem on production mode runs or builds with no errors).

since : 
1  - i have no output whatsoever  . 
2 - i've been working on this project for a long time .   
i have no idea how to debug this or reproduce the problem in a small sample of code. 

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

managed

### What platform(s) does this occur on?

Android, iOS

### SDK Version (managed workflow only)

43

### Environment

  Expo CLI 4.12.1 environment info:
    System:
      OS: Linux 5.14 Arch Linux
      Shell: 5.8 - /bin/zsh
    Binaries:
      Node: 16.11.1 - /usr/bin/node
      npm: 8.0.0 - /usr/bin/npm
    npmPackages:
      expo: ^43.0.0 => 43.0.3 
      react: 17.0.1 => 17.0.1 
      react-dom: 17.0.1 => 17.0.1 
      react-native: 0.64.3 => 0.64.3 
      react-native-web: 0.17.1 => 0.17.1 
    npmGlobalPackages:
      expo-cli: 4.12.1
    Expo Workflow: managed

### Reproducible demo or steps to reproduce from a blank project

the reason why i'm not able to do this is written in the summery...

### UPDATE : 
 - i seen to get the white screen after the splash screen on ios and i seem to get the cover screen (the screen expo uses to build my project) again after the splash screen on android -and it just gets stuck there- 
 - also it seems so if i remove some UI elements from the Home Screen the rest of the app renders but the portion i'm removing is like the main point of the home page (the widgets are the only thing that can stay)
still though without logs it's really really hard to debug the app  + why do we even have expo start --no-dev --minify if we're not getting logs anyways ? ... it really is just like a standalone build (as in not easy to debug on ?)

**Comment by ayman-rahmon** [comment]
additional Note : 

when i run the app and then switch to the production version after the home screen is loaded no problems happen then...  so it might be related to the launching process... app launcher ? maybe ?  ... 


this is just a speculation.  idk that for a fact i'm just stating what's happening with me

**Comment by ayman-rahmon** [comment]
hello again , 

i just tried ejecting to the bare workflow and build for android with eas and got the following error


```
`Running npm in the root dir of your repository 

[stderr] npm

[stderr] WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.

[stderr] npm

[stderr] WARN deprecated core-js@1.2.7: core-js@<3.3 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Please, upgrade your dependencies to the actual version of core-js.

[stderr] npm WARN deprecated sane@4.1.0: some dependency vulnerabilities fixed, support for node < 10 dropped, and newer ECMAScript syntax/features added

[stderr] npm WARN deprecated uglify-es@3.3.9: support for ECMAScript is superseded by `uglify-js` as of v3.13.0

[stderr] npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated

[stderr] npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated

[stderr] npm ERR! code

[stderr] E404

[stderr] npm ERR! 404 Not Found - GET http://npm-cache-service.production.svc.cluster.local:4873/- - File not found

[stderr] npm ERR! 404

[stderr] npm ERR! 404  '-@^0.0.1' is not in the npm registry.

[stderr] npm

[stderr] ERR! 404 You should bug the author to publish it (or use the name yourself!)

[stderr] npm ERR! 404 It was specified as a dependency of 'build'

[stderr] npm ERR! 404 

[stderr] npm ERR! 404 Note that you can also install from a

[stderr] npm ERR! 404 tarball, folder, http url, or git url.

[stderr] npm

[stderr] ERR! A complete log of this run can be found in:

[stderr] npm ERR!     /root/.npm/_logs/2021-11-18T04_02_08_045Z-debug.log

npm exited with non-zero code: 1`

```

i still don't even know where to begin to work with this since it's telling me to bug the author ??  weird error message...

**Comment by ayman-rahmon** [comment]
it turned out to be a problem with some ui (which wasn't giving me any problem in development mode but broke everything in production mode) so i just removed that and replaced it and now the app is working but i still don't know why there are no logs when we use expo start --no-dev --minify (it's as if the build is production sure but we need some output to debug still - since it's not stand alone app-)


i'm closing this issue now , i may or may not open another ticket for the logs thing... though...

**Comment by dlamon1** [comment]
@ayman-rahmon What was the ui issue? I am having the same issue and cannot track it down, no matter what I remove from the project.

**Comment by ayman-rahmon** [comment]
I'm sorry for the late response and I'm very sorry that you're having this issue.


the issue was that some third party component that i used in the main screen back then caused this weird behavior for some reason. I assumed the problem happened because development mode runs on NPM and all packages that run on NPM work just fine. but once you move to the production mode you can actually see what will be packaged with the app and what can only run on NPM. 


this is a huge problem when using EXPO in my opinion ... you can't really know what will work in production mode and what can only run when supported by NPM in development mode. (The same issue happened with me when i tried to use sockets. it didn't even matter if i used Socket.io or if i used web sockets directly they just are not supported to work with expo, at least at the time they weren't ).

**Comment by codehippie1** [comment]
@ayman-rahmon Do you know which third-party package was causing the issue? Just ran into this now.

**Comment by ayman-rahmon** [comment]
If this question had been raised at the time, I would have been able to identify the exact package that caused the issue. Unfortunately, I left that company two years ago and no longer have access to the source code.

I hope this helps: back then, I started my troubleshooting by removing one view at a time in production mode to see if the issue persisted. Eventually, I found the exact view causing the problem. Within that view, I disabled one component at a time until the issue was resolved, which allowed me to identify the problematic package.
